cores/arduino/ard_sup/gpio/ap3_gpio.cpp

```diff
@@ -125,6 +137,10 @@ extern void digitalWrite(uint8_t pin, uint8_t val)
     {
         return;
     }
+    if (ap3_get_funct_sel(pad) != GPIO_FUNCTION)
+    {
+        pinMode(pin, OUTPUT);
+    }
```

Comment:
Looks good. As we discussed the downside is that a pin with a different mode could accidentally be deconfigured by any stray call to ```digitalWrite```. Then, also, the typical *need* to set pinMode(OUTPUT) before digitalWrite is technically not there - but it won't break compatibility with existing code. Just could permit bad habits. 

So... the real judge of whether we accept this PR is whether we think that the need to call digitalWrite right after analogWrite is important enough to counteract those cons. I am tempted to leave that judgement to Nate, but if he prefers I would make a decision.